root/app/Modules/GoogleCalendarOperation.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Modules;
+
+use Illuminate\Support\Carbon;
+use Spatie\GoogleCalendar\Event;
+
+class GoogleCalendarOperation
+{
+    public function __construct()
+    {
+    }
+
+    public function getFutureSchedule()
+    {
+        $events = Event::get(); // 未来の全イベントを取得する
+
+        // 今後の予定を取得する処理を実行する
+        $futureSchedule = [];
+        foreach ($events as $event) {
+            $futureSchedule[] = [
+                'id' => $event->id,
+                'name' => $event->name,
+                'description' => $event->description,
+                'startDateTime' => $event->startDateTime,
+                'endDateTime' => $event->endDateTime,
+            ];
+        }
+        return $futureSchedule;
+    }
+
+    public function createSchedule(Object $scheduleData) {
+        $date = new Carbon($scheduleData->date);
+
+        $event = new Event;
+        $event->name = $scheduleData->title;
+        $event->startDateTime = $date;
+        $event->endDateTime = $date->addHour();   // 1時間後
```

Comment:
[MUST]
Carbonはミュータブルのため、startDateTimeとendDateTimeは同じオブジェクトが格納されます。

CarbonImmutableを使うか、クローンしたものを1時間プラスする必要があります。

```suggestion
        $event->startDateTime = $date;
        $event->endDateTime = $date->copy()->addHour();   // 1時間後
```

参考
https://epkotsoftware.github.io/training/programming/#%E3%83%9F%E3%83%A5%E3%83%BC%E3%82%BF%E3%83%96%E3%83%AB%E3%82%A4%E3%83%9F%E3%83%A5%E3%83%BC%E3%82%BF%E3%83%96%E3%83%AB